language/move-lang/functional-tests/tests/epilogue/out_of_gas_in_nested_module.move

```diff
@@ -0,0 +1,34 @@
+module Swapper {
+    use 0x1::Vector;
+    public fun call(x: u64) {
+        swap_it_up(x)
+    }
+    public fun swap_it_up(vec_len: u64) {
+        let v = Vector::empty();
+
+        let i = 0;
+        while (i < vec_len) {
+          Vector::push_back(&mut v, i);
+          i = i + 1;
+        };
+
+        i = 0;
+
+        while (i < vec_len / 2) {
+            Vector::swap(&mut v, i, vec_len - i - 1);
```

Comment:
Lol, did you write the test this way because `swap` has a high gas cost?

Reply:
I did indeed!